### PR TITLE
Support AUC on CV for classification problem

### PR DESCRIPTION
**Pull request by oooo26**
- Use (negative) AUC as CV score for `LogisticRegression` and `MultinomialRegression`;
- Both [OvO and OvR Algorithm](https://scikit-learn.org/stable/modules/model_evaluation.html#multi-class-case) are implemented for multinomial case;
- Speed and performance have been tested.

### CODE REVIEW

**Comment by oooo26** [comment]
Besides, I think `ic_type` can be renamed as `eval_type` because this argument is not only for IC now.
And shall we add an argument to specify OvO/OvR for `MultinomialRegression`?

**Comment by Mamba413** [comment]
@oooo26 ，shall we change `ic_type` to `eval_type`?

**Comment by Mamba413** [comment]
> @oooo26 ，shall we change `ic_type` to `eval_type`?

may be other. you can search `scikit-learn` for reference

**Comment by oooo26** [comment]
> may be other. you can search `scikit-learn` for reference

Oh Yes. They have a `scoring` argument in [LogisticRegressionCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegressionCV.html) to control it, but not in the non-CV one. They don't seem to offer different IC type for classification too, only for [Lasso](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoLarsIC.html#sklearn.linear_model.LassoLarsIC).

Shall we add a `cv_score` or something? (Only `scoring` might be misunderstood since our regressor combines CV and non-CV problem.)

**Comment by oooo26** [comment]
And I think they can share the same API in Cpp (that's what I have done so far), since only one of the criteria will be used.

**Comment by Mamba413** [comment]
two excellent comments. just do it.

**Comment by Mamba413** [comment]
@ EQUIWDH you can inspect cpp code and program  for cindex similarly.

**Comment by oooo26** [comment]
> two excellent comments. just do it.

OK! I will update soon.

**Comment by oooo26** [comment]
Add `cv_scoring`:
- `cv_scoring='test_loss'`: default
- `cv_scoring='roc_auc'`: additional for logistic
- `cv_scoring='roc_auc_ovo'` or `'roc_auc_ovr'`: additional for multinomial

Since AUC score seems to perform better on Logistic (both speed and accuracy), maybe we can set `LogisticRegression(cv_score='roc_auc')` as default.

**Comment by oooo26** [comment]
Oh there is a small conflict on R: when using CV, the `ic_coef` should be set as 0 so that it can use test loss as before.

But currently, it is 1. We may need to change it in `R-package/R/utility.R`. (Otherwise, there will be some annoying warnings...)

**Comment by Mamba413** [comment]
I skim the R code in R-package now, and I think @bbayukari can quickly address this remaining warning in `abesspca` (see https://github.com/abess-team/abess/actions/runs/4044489654/jobs/6954695218).

**Comment by oooo26** [comment]
> I skim the R code in R-package now, and I think @bbayukari can quickly address this remaining warning in `abesspca` (see https://github.com/abess-team/abess/actions/runs/4044489654/jobs/6954695218).

Yep, just setting `ic_type=0` will fix it. Because `ic_type=1` will represent AUC when using CV now.

**Comment by Mamba413** [comment]
> > I skim the R code in R-package now, and I think @bbayukari can quickly address this remaining warning in `abesspca` (see https://github.com/abess-team/abess/actions/runs/4044489654/jobs/6954695218).
> 
> Yep, just setting `ic_type=0` under CV will fix it. Because `ic_type=1` will represent AUC score for logistic now.

Seems everything is fine now! You can merge it if you also believe this is fine.
